### ISSUE TITLE
skip / remove ` ModLoaderStore.get("any_mod_hooked")` check in export plugin

### ISSUE THREAD

**Issue by KANAjetzt** [GodotModding/godot-mod-loader#487]
>               I think the` ModLoaderStore.get("any_mod_hooked")` kills all the performance-benefits of checking ModLoaderStore.any_mod_hooked. 
> This was originally a way to reduce the modding overhead when running a modless version of a game. 
> 2 topics:
>  - can we somehow skip the ` ModLoaderStore.get("any_mod_hooked")`  call?
>  - can we skip the check altogether when dynamically generating the scripts, since there we only create the methods when hooks actually exist.
>  
>  (Same below on l.271)
> 
> _Originally posted by @pirey0 in https://github.com/GodotModding/godot-mod-loader/pull/408#discussion_r1832930546_

> [!NOTE]
>   #429 Removed this check from dynamic hook creation.

In addition to the mentioned performance issue, we currently have the following Parse error:
`SCRIPT ERROR: Parse Error: Not all code paths return a value.`

Example:
```gdscript
func check_sight(target_position: Vector3) -> bool:
	if ModLoaderStore.get("any_mod_hooked") and ModLoaderStore.any_mod_hooked:
		return _ModLoaderHooks.call_hooks(vanilla_608418131_check_sight, [target_position], 1845897103)
```

**Comment by KANAjetzt** [comment]
This issue only occurs with `@tool` scripts in the editor. For the export plugin, a simple `if ModLoaderStore.any_mod_hooked:` is sufficient. This means we can resolve it by adding the additional `get()` check only when processing scripts with the development tool.

_\*The parser error still needs to be fixed though_

**Comment by KANAjetzt** [comment]
closed by #498
